plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -348,7 +349,7 @@ public IcebergTableHandle getTableHandle(
         }
         else {
             tableSnapshotId = Optional.ofNullable(table.currentSnapshot()).map(Snapshot::snapshotId);
-            tableSchema = table.schema();
+            tableSchema = getCurrentSnapshotSchema(table);
```

Comment:
In https://github.com/trinodb/trino/pull/13704/files#r969206216 you're showing that this works, but i don't understand yet why it does so.

What's the situation when `table.schema()` is different from `getCurrentSnapshotSchema(table)`?

In that situation, how do they differ?

Reply:
The situation is after rollback. In the last SELECT statement in `testRollbackSnapshotWithSchemaEvolution`, 
`table.schema()` returns
```
table {
  1: col0: optional int
  2: col1: optional int
}
```

`getCurrentSnapshotSchema(table)` returns
```
table {
  1: col0: optional int
}
```